Implementation work with bluetooth through Windows API

nk0leg opened the pull request:
https://adc.luxoft.com/jira/browse/SDLWIN-121

nk0leg commented:
please review @malirod, @anosach-luxoft , @pestOO

AGaliuzov commented on src/components/transport_manager/include/transport_manager/bluetooth/bluetooth_device_scanner.h:
```diff
@@ -133,7 +133,7 @@ class BluetoothDeviceScanner : public DeviceScanner {
    */
 #ifdef OS_WINDOWS
   std::vector<RfcommChannelVector> DiscoverSmartDeviceLinkRFCOMMChannels(
```
The same vector uses below as well. So maybe it worth to avoid duplication

nk0leg replied:
No it' not duplicate. It's different function.

AGaliuzov replied:
Compare lines 135 and 138

AGaliuzov commented on src/components/transport_manager/src/bluetooth/bluetooth_device_scanner_win.cc:
```diff
@@ -224,68 +128,47 @@ void BluetoothDeviceScanner::UpdateTotalDeviceList() {
+  hbf = BluetoothFindFirstDevice(&bdsp, &bdi);
+  if (hbf) {
+	  do {
+		  found_devices.push_back(bdi);
```
If you use the vector here you don't need number_of_devices variable anymore

nk0leg replied:
variable number_of_devices  uses in a debug message. line 164.

AGaliuzov replied:
But you can use vector size instead

malirod replied:
vector `empty`

AGaliuzov commented on src/components/transport_manager/src/bluetooth/bluetooth_device_scanner_win.cc:
```diff
@@ -360,80 +243,65 @@ const std::vector<BTH_ADDR*>& device_addresses) {
-		LOG4CXX_ERROR(logger_, "BtServiceSearch(): WSALookupServiceBegin() failed with error code ", WSAGetLastError());
 	}
 
+  channels->push_back(1); //temporary work around 
```
Workaround for what?

nk0leg replied:
Please look pool request description.

AGaliuzov commented on src/components/transport_manager/src/bluetooth/bluetooth_transport_adapter_win.cc:
```diff
@@ -156,8 +146,17 @@ bool BluetoothTransportAdapter::Restore() {
+	btdi.fAuthenticated = false;
+
+	BluetoothDevice* bluetooth_device = 
+			new BluetoothDevice(btdi, name.c_str(), rfcomm_channels);
```
Why you need to create pointer twice. I would rather use makeShared function

nk0leg replied:
Where is twice pointers?

AGaliuzov replied:
First where you create raw pointer and second when you create shared from this raw

nk0leg replied:
It's original GENEVI logic.

AGaliuzov replied:
So you can change it

AGaliuzov commented on src/components/transport_manager/src/bluetooth/bluetooth_device_scanner_win.cc:
```diff
@@ -360,80 +243,65 @@ const std::vector<BTH_ADDR*>& device_addresses) {
+
+	HANDLE hLookup;
+	int result = WSALookupServiceBegin(&querySet, flags, &hLookup);
+	if (result == 0) {
```
Useless condition

malirod commented:
@nk0leg, please, use small commits. Big commits with many unrelated changes is the svn style. It's hard to read, understand and navigate such commits.

Commit:
Implementation work with bluetooth through Windows API

In this solution yet not implemented discovering, search and parsing SDP(service device protocol) data
i.e. service-uuid and RFCOMM remote port for connection to service.
Because current bluetooth driver not contained this info.

https://adc.luxoft.com/jira/browse/SDLWIN-121

anosach-luxoft commented on src/appMain/CMakeLists.txt:
```diff
@@ -162,7 +162,7 @@ set (HEADERS
 if(${CMAKE_SYSTEM_NAME} MATCHES "Windows")
   if(NOT QT_PORT)
-    list(APPEND LIBRARIES sqlite3 bthprops)
+    list(APPEND LIBRARIES sqlite3 bthprops Ws2_32)
```
Do we need to link this library in case BUILD_BT_SUPPORT OFF?

nk0leg replied:
Yes. Ws2_32.lib  - library needed for winsock

malirod replied:
We were using windows sockets before. Will it it work without this lib?

anosach-luxoft replied:
How it works now?

nk0leg replied:
in the files: src\components\transport_manager\src\tcp\tcp_client_listener.cc :36
src\components\transport_manager\src\tcp\tcp_device.cc: 38
src\components\transport_manager\src\tcp\tcp_socket_connection.cc :43
yet included "#pragma comment(lib, "Ws2_32.lib")"

malirod replied:
Maybe it worth do remove from there too?

nk0leg replied:
Yes. Previous must be removed.
Enough  it's inclusion.

anosach-luxoft commented:
@nk0leg, please look throw your changes one more time and fix all indents
